gradle/libs.versions.toml

```diff
@@ -96,9 +96,10 @@ mockito-core = { group = "org.mockito", name = "mockito-core", version.ref = "mo
 hive2-metastore = { group = "org.apache.hive", name = "hive-metastore", version.ref = "hive2"}
 hive2-exec = { group = "org.apache.hive", name = "hive-exec", version.ref = "hive2"}
 hive2-common = { group = "org.apache.hive", name = "hive-common", version.ref = "hive2"}
-hadoop2-hdfs = { group = "org.apache.hadoop", name = "hadoop-hdfs", version.ref = "hadoop2" }
-hadoop2-common = { group = "org.apache.hadoop", name = "hadoop-common", version.ref = "hadoop2"}
-hadoop2-mapreduce-client-core = { group = "org.apache.hadoop", name = "hadoop-mapreduce-client-core", version.ref = "hadoop2"}
+hadoop3-hdfs = { group = "org.apache.hadoop", name = "hadoop-hdfs", version.ref = "hadoop3" }
+hadoop3-common = { group = "org.apache.hadoop", name = "hadoop-common", version.ref = "hadoop3"}
+hadoop3-client = { group = "org.apache.hadoop", name = "hadoop-client", version.ref = "hadoop3"}
+hadoop3-mapreduce-client-core = { group = "org.apache.hadoop", name = "hadoop-mapreduce-client-core", version.ref = "hadoop3"}
```

Comment:
I think we can both keep hadoop2 and hadoop3, and only use hadoop3 in `catalog-hadoop` module.

I was worrying if there's a potential issue in `catalog-hive` to upgrade to hadoop3. So I don't suggest to upgrade to Hadoop 3 everywhere.

With the fix of #2397 , the classpath is isolated for different ITs, so it is ok to keep two hadoop versions in different module.

Reply:
okay, I will make some adjustments, but we also need to use hadoop3 in `catalog-lakehouse-iceberg` module.

Reply:
>  but we also need to use hadoop3 in catalog-lakehouse-iceberg module

Why do we require hadoop3 in Iceberg, I cannot get it, can you please explain more when you have time?

Reply:
I originally thought that Iceberg catalog supports hadoop backend. But later found that the hadoop backend is not currently supported, which is required for Paimon catalog within Xiaomi.

At the same time, seeing the reply from FANNG1, I will reconfirm the details. Thx.

Reply:
Hadoop catalog in Iceberg is mainly for test/simple use,  it is seldom used in the production environment.

@YxAc Since Iceberg's hive catalog backend relies on Hadoop. I would not suggest upgrading Hadoop in the Iceberg catalog either. WDYT?

Reply:
@jerryshao In the long run, I think if Gravitino supports lock, thus natively supporting data lakes, iceberg catalog will also need to upgrade to hadoop3.

But if you are more cautious about this change, i can modify it as needed in the future. Can you reconfirm it? Thx.

Reply:
Thanks @YxAc , I'm conservative to upgrade everything to hadoop 3 for now. Because when I did Spark's Hadoop 3 upgrading, there're several potential issues besides just updating the version number. And currently, we still don't know if there's an issue in it, so I would suggest to keep the changes minimal. Later on, when we verified everything, we can bump the whole version to Hadoop 3.